# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: modules/libjar/nsJARInputStream.h
# Commit: b07887474f51
# Full Hash: b07887474f51c5d5e01cb0188c18c8618c77be57
# Author: Doug Thayer <dothayer@mozilla.com>
# Date: 2020-07-02 03:58:28
# Regressor Bug: 1627075
# File Overlap Count: 1
# Description:
#   Bug 1627075 - Route Omnijar requests through StartupCache r=froydnj
#   
#   This should be a relatively straightforward patch. Essentially, we implement
#   a wrapper class (and friends) around nsZipArchive (and friends), which transparently
#   caches entries from the underlying zip archive in the StartupCache. This will break
# ==============================================================================

diff -r 65c0e6790a33 -r b07887474f51 modules/libjar/nsJARInputStream.h
--- a/modules/libjar/nsJARInputStream.h	Wed Jul 01 17:55:38 2020 +0000
+++ b/modules/libjar/nsJARInputStream.h	Wed Jul 01 17:09:53 2020 +0000
@@ -11,6 +11,7 @@
 #include "nsJAR.h"
 #include "nsTArray.h"
 #include "mozilla/Attributes.h"
+#include "mozilla/Omnijar.h"
 
 #ifdef MOZ_JAR_BROTLI
 struct BrotliDecoderStateStruct;
@@ -51,11 +52,11 @@
  private:
   ~nsJARInputStream() { Close(); }
 
-  RefPtr<nsZipHandle> mFd;  // handle for reading
-  uint32_t mOutSize;        // inflated size
-  uint32_t mInCrc;          // CRC as provided by the zipentry
-  uint32_t mOutCrc;         // CRC as calculated by me
-  z_stream mZs;             // zip data structure
+  mozilla::CacheAwareZipHandle mItemHandle;  // handle for reading
+  uint32_t mOutSize;                         // inflated size
+  uint32_t mInCrc;                           // CRC as provided by the zipentry
+  uint32_t mOutCrc;                          // CRC as calculated by me
+  z_stream mZs;                              // zip data structure
 #ifdef MOZ_JAR_BROTLI
   BrotliDecoderStateStruct* mBrotliState;  // Brotli decoder state
 #endif